src/cosmospy/_transaction.py

```diff
@@ -21,87 +26,149 @@ class Transaction:
+            fee: int,
+            gas: int,
+            fee_denom: str = "uatom",
+            fee_account: str = "",
+            memo: str = "",
+            chain_id: str = "cosmoshub-4",
+            hrp: str = DEFAULT_BECH32_HRP,
+            sync_mode: SyncMode = "broadcast_tx_sync",
     ) -> None:
         self._privkey = privkey
         self._account_num = account_num
         self._sequence = sequence
         self._fee = fee
         self._fee_denom = fee_denom
+        self._fee_account = fee_account
         self._gas = gas
-        self._memo = memo
         self._chain_id = chain_id
         self._hrp = hrp
         self._sync_mode = sync_mode
-        self._msgs: list[dict] = []
-
-    def add_transfer(self, recipient: str, amount: int, denom: str = "uatom") -> None:
-        transfer = {
-            "type": "cosmos-sdk/MsgSend",
-            "value": {
-                "from_address": privkey_to_address(self._privkey, hrp=self._hrp),
-                "to_address": recipient,
-                "amount": [{"denom": denom, "amount": str(amount)}],
-            },
-        }
-        self._msgs.append(transfer)
-
-    def get_pushable(self) -> str:
-        pubkey = privkey_to_pubkey(self._privkey)
-        base64_pubkey = base64.b64encode(pubkey).decode("utf-8")
-        pushable_tx = {
-            "tx": {
-                "msg": self._msgs,
-                "fee": {
-                    "gas": str(self._gas),
-                    "amount": [{"denom": self._fee_denom, "amount": str(self._fee)}],
-                },
-                "memo": self._memo,
-                "signatures": [
-                    {
-                        "signature": self._sign(),
-                        "pub_key": {"type": "tendermint/PubKeySecp256k1", "value": base64_pubkey},
-                        "account_number": str(self._account_num),
-                        "sequence": str(self._sequence),
-                    }
-                ],
-            },
-            "mode": self._sync_mode,
-        }
-        return json.dumps(pushable_tx, separators=(",", ":"))
+        self._tx_body = tx.TxBody()
+        self._tx_body.memo = memo
+        self._tx_raw = tx.TxRaw()
 
-    def _sign(self) -> str:
-        message_str = json.dumps(self._get_sign_message(), separators=(",", ":"), sort_keys=True)
-        message_bytes = message_str.encode("utf-8")
+    def add_transfer(
+            self, recipient: str, amount: int, denom: str = "uatom", hrp: str = DEFAULT_BECH32_HRP
+    ) -> None:
+        msg = transfer.MsgSend()
+        msg.from_address = privkey_to_address(self._privkey, hrp=hrp)
+        msg.to_address = recipient
+        _amount = coin.Coin()
+        _amount.denom = denom
+        _amount.amount = str(amount)
+        msg.amount.append(_amount)
+        msg_any = Any.Any()
+        msg_any.Pack(msg)
+        msg_any.type_url = "/cosmos.bank.v1beta1.MsgSend"
+        self._tx_body.messages.append(msg_any)
+
+    def add_ibc_transfer(
```

Comment:
This seems like new API/functionality unrelated to the deprecated endpoint. Can we design, add and test it in a separate PR?

Reply:
Yes, we can do that too.  We then need to also change the tests because I already added a test for this tx type.